src/core/mtu_discovery.h

```diff
@@ -0,0 +1,69 @@
+/*++
+
+    Copyright (c) Microsoft Corporation.
+    Licensed under the MIT License.
+
+--*/
```

Comment:
Lots of comments needed in this file please

Reply:
Like where? I tried to explain everything that was algorithm based. The only way to comment it more is to literally comment lines of code. (I already added the abstract locally)